### PR TITLE
Refactor sidecar to unified task model

### PR DESCRIPTION
**Pull request by bdchatham**
## Summary

- Replaces the standalone scheduler with in-engine scheduled task support. Tasks are now assigned a UUID on submission and tracked through a unified `TaskResult` model, whether one-shot or recurring.
- Removes `Status` field from `TaskResult` (the presence of `Error` implies success/failure). `Schedule` is now a struct with `Cron` and `BlockHeight` fields for future extensibility. Cron validation moves to the server API layer.
- All endpoints move under `/v0/` prefix with `POST /v0/tasks` (202/201), `GET /v0/tasks`, `GET /v0/tasks/{id}`, `DELETE /v0/tasks/{id}`. `StatusResponse` simplified to only return node lifecycle status.

## Test plan

- [x] All sidecar engine tests pass (submit, scheduled, eval, bounded history)
- [x] All server tests pass (routes, validation, status codes)
- [x] Cron validation tests pass
- [x] Corresponding `sei-sidecar-client-go` OpenAPI spec and client updated and pushed to main

### CODE REVIEW

File: sidecar/engine/cron_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	if err := ValidateCron("not a cron"); err == nil {
+		t.Fatal("expected error for invalid cron")
+	}
+}
+
+func TestValidateCronAcceptsValid(t *testing.T) {
+	if err := ValidateCron("*/5 * * * *"); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+}
```
Comment (masih) [inline sidecar/engine/cron_test.go:32]: Turn all three of these tests into 1 table test? That should make it trivial to add new test cases and would reduce duplicate code.

File: sidecar/engine/engine.go [hunk 2]
```diff
@@ -8,80 +8,77 @@ import (
...
 	go e.loop()
 	return e
 }
 
-// loop is the serial worker that processes non-scheduled tasks one at a time.
-// After each task completes it stores the result and releases the task lock.
-// Exits when the engine context is cancelled.
+// loop is the serial worker that processes one-shot tasks one at a time.
 func (e *Engine) loop() {
 	for {
```
Comment (masih) [inline sidecar/engine/engine.go]: condition this loop with e.ctx err being nil

File: sidecar/engine/engine.go [hunk 3]
```diff
@@ -90,35 +87,99 @@ func (e *Engine) loop() {
...
+	e.mu.Unlock()
+
+	return id, nil
 }
 
-// execute runs a handler synchronously and logs the outcome. Returns the
-// handler error (if any) so the caller can record it.
+// EvalSchedules checks all scheduled tasks and fires any that are due.
+// Scheduled tasks run in their own goroutines, bypassing the one-shot lock.
+func (e *Engine) EvalSchedules() {
```
Comment (masih) [inline sidecar/engine/engine.go:138]: Should this advance `NextRunAt` or mark the task as "in flight" - as is I _think_ reoccurring tasks can be triggered multiple times concurrently while the old runs are still running.

File: sidecar/engine/engine.go [hunk 4]
```diff
@@ -8,80 +8,77 @@ import (
 	"sync"
 	"sync/atomic"
 	"time"
+
+	"github.com/google/uuid"
 )
 
-// ErrBusy is returned by Submit when a non-scheduled task is already running.
+const maxResults = 5
```
Comment (masih) [inline sidecar/engine/engine.go]: 5 seems too small considering the memory footprint. Bump this to 10?

File: sidecar/engine/engine.go [hunk 5]
```diff
@@ -8,80 +8,77 @@ import (
...
-			err := e.execute(env.taskType, env.handler, env.params)
+			err := e.execute(env.task.Type, env.handler, env.task.Params)
 
-			result := &TaskResult{Type: string(env.taskType)}
+			now := time.Now()
+			result := &TaskResult{
+				ID:          env.id,
+				Type:        string(env.task.Type),
+				Params:      env.task.Params,
+				SubmittedAt: now,
```
Comment (masih) [inline sidecar/engine/engine.go]: This records completion time not submission time. I would either remove it to avoid confusion or correct it to record what it claims to record.

**Review by masih — CHANGES_REQUESTED** [review]
One blocker on concurrent reoccurrence of tasks. Left some suggestions. Otherwise LGTM 🙌

File: sidecar/engine/engine.go (hunk 3 above)
Comment (bdchatham) [reply sidecar/engine/engine.go:138]: Yeah, this was written assuming only serial execution by the engine. Added a defensive check

**Review by masih — APPROVED** [review]

File: sidecar/engine/engine.go (hunk 3 above)
Comment (bdchatham) [reply sidecar/engine/engine.go:138]: I'd gotten turned around -- JVM warming up. This was definitely an issue
